src/S3IteratorLibsvm.cpp

```diff
@@ -241,6 +263,7 @@ bool S3IteratorLibsvm::buildDatasetLibsvm(
     return true;
   } catch (...) {
     // readNum throws exception if it can't find a digit right away
+    std::cerr << "Error parsing" << std::endl;
     return false;
```

Comment:
If there is an error parsing, wouldn't it be better to throw a runtime error instead of stating that there is an error and returning? Returning false will have the code think that it has reached the end of the file. I understand that the part of the dataset that has been parsed will be valid, but it's possible that the code is cutoff in the middle of a sample, which is an undesirable behavior.